### PR TITLE
[CLIPPER-73] CIFAR SSH-only demo

### PR DESCRIPTION
**Pull request by dcrankshaw**

### CODE REVIEW

**Comment by Corey-Zumar** [comment]
Password-protected SSH keys result in the user being prompted for password input when the Clipper Manager is instantiated. This results in a Jupyter Notebook crash resulting from an incompatible call to python's `get_pass()` method.

**Comment by dcrankshaw** [comment]
@Corey-Zumar I think this is working again

File: examples/cifar_demo/rise_demo_frontend_dev.ipynb
Comment (Corey-Zumar) [inline examples/cifar_demo/rise_demo_frontend_dev.ipynb:150]: Can we enumerate these arguments as key-value pairs for user clarity?
`selection_policy="EXP4"`, etc...

File: examples/cifar_demo/rise_demo_data_scientist.ipynb
Comment (Corey-Zumar) [inline examples/cifar_demo/rise_demo_data_scientist.ipynb:105]: enumerate these arguments as key-value pairs for user clarity

File: examples/cifar_demo/rise_demo_data_scientist.ipynb
Comment (Corey-Zumar) [inline examples/cifar_demo/rise_demo_data_scientist.ipynb:170]: enumerate these arguments as key-value pairs for user clarity

File: examples/cifar_demo/rise_demo_data_scientist.ipynb
Comment (Corey-Zumar) [inline examples/cifar_demo/rise_demo_data_scientist.ipynb]: We should remove inline TODOs from the demo

File: examples/cifar_demo/rise_demo_data_scientist.ipynb
Comment (Corey-Zumar) [inline examples/cifar_demo/rise_demo_data_scientist.ipynb]: Remove inline TODO

File: containers/python/TensorFlowCifarDockerfile
Comment (Corey-Zumar) [inline containers/python/TensorFlowCifarDockerfile]: Why are there 3 TODOs here?

File: containers/python/SklearnCifarDockerfile
Comment (Corey-Zumar) [inline containers/python/SklearnCifarDockerfile]: Three TODOs?

File: examples/cifar_demo/README.md
Comment (Corey-Zumar) [inline examples/cifar_demo/README.md:10]: If we can get https://github.com/ucbrise/clipper/pull/60 merged, we should include information about running the demo locally. This seems like a very critical addition, as there is arguably much less overhead to installing Docker locally than launching AWS instances.

**Review by Corey-Zumar — CHANGES_REQUESTED** [review]
Definitely an improvement in that the demo works as intended with the EXP4 selection policy. However, the following issues remain:
- The `get_pass()` issue is still occurring when attempting to use a password-protected SSH key with Jupyter Notebook 4.0.6
- If we expect users to experiment with different selection policies, the demo does not have compatible CIFAR binary prediction for EXP3, UCB, or EpsilonGreedy. Disregard this point if it's a nonissue for the demo.
- Nit: there's an extra, empty cell at the bottom of the `fronted_dev` notebook

I've left several other comments to address as well - most of them are minor

**Comment by dcrankshaw** [comment]
I can't reproduce the `get_pass()` issue. It works fine with Jupyter 4.2.1. Given time constraints I'm inclined to ignore it for now.

File: examples/cifar_demo/rise_demo_frontend_dev.ipynb
Comment (dcrankshaw) [reply examples/cifar_demo/rise_demo_frontend_dev.ipynb:150]: That's what it was originally. But the only way to do that while also still having them be required is a hack with kwargs that I thought looked uglier. I think you're probably right though, I'm changing it back.

File: examples/cifar_demo/rise_demo_data_scientist.ipynb
Comment (dcrankshaw) [reply examples/cifar_demo/rise_demo_data_scientist.ipynb]: These TODOs need to be addressed at the very end. `latest` is the docker build from the develop branch so it will break the demo until the PR gets merged. Holding off until we get everything else addressed.

File: examples/cifar_demo/README.md
Comment (dcrankshaw) [reply examples/cifar_demo/README.md:10]: Sounds good, but let's do that in the PR for #60.

**Comment by dcrankshaw** [comment]
Fixed the issue preventing the other selection policies from working. Problem was [here](https://github.com/dcrankshaw/clipper/blob/cifar_ec2_demo/src/libclipper/src/selection_policies.cpp#L149). Changed it to return 0.0 in case of no predictions.

File: examples/cifar_demo/rise_demo_frontend_dev.ipynb
Comment (dcrankshaw) [reply examples/cifar_demo/rise_demo_frontend_dev.ipynb:150]: Hmm actually on second thought it makes the clipper_manager library much more confusing. [This StackOverflow answer](http://stackoverflow.com/a/5940228/814642) describes what I was doing before, and I think having all the functions just take kwargs is confusing. In the tutorial I'll make sure to explain what the arguments are.

File: containers/python/SklearnCifarDockerfile
Comment (dcrankshaw) [reply containers/python/SklearnCifarDockerfile]: Removed

File: containers/python/TensorFlowCifarDockerfile
Comment (dcrankshaw) [reply containers/python/TensorFlowCifarDockerfile]: Removed

File: examples/cifar_demo/rise_demo_data_scientist.ipynb
Comment (dcrankshaw) [reply examples/cifar_demo/rise_demo_data_scientist.ipynb]: Removed

File: examples/cifar_demo/rise_demo_data_scientist.ipynb
Comment (dcrankshaw) [reply examples/cifar_demo/rise_demo_data_scientist.ipynb]: Removed

File: management/clipper_manager.py
Comment (dcrankshaw) [inline management/clipper_manager.py]: removed todo

File: management/clipper_manager.py
Comment (dcrankshaw) [inline management/clipper_manager.py]: removed todo

**Comment by dcrankshaw** [comment]
@Corey-Zumar addressed your comments. Note that the demo won't work until it gets merged because I changed a bunch of docker files to from `test` to `latest`. This changes them from pulling the test version of the docker images (the ones corresponding to the code in this PR) to pulling the docker images from the code on `develop`.

File: examples/cifar_demo/README.md
Comment (Corey-Zumar) [reply examples/cifar_demo/README.md:10]: That makes more sense

File: examples/cifar_demo/rise_demo_frontend_dev.ipynb
Comment (Corey-Zumar) [reply examples/cifar_demo/rise_demo_frontend_dev.ipynb:150]: Sounds good, as long as the user is aware of the role of each argument

**Comment by Corey-Zumar** [comment]
LGTM. I'll file an issue for `get_pass()` and see if I can find a fix and a way of reliably replicating it on another system.

**Review by Corey-Zumar — APPROVED** [review]

**Pull request merged**
